embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/KafkaDataFormatsTest.java

```diff
@@ -0,0 +1,727 @@
+  }
+
+  @Test
+  public void test_indexKafka_protobufDataFormat()
+  {
+    kafkaServer.createTopicWithPartitions(dataSource, 3);
+    EventSerializer serializer = new ProtobufEventSerializer();
+
+    int recordCount = generateStreamAndPublishToKafka(dataSource, serializer, false);
+
+    FileBasedProtobufBytesDecoder protobufBytesDecoder = new FileBasedProtobufBytesDecoder(
+        "data/protobuf/wikipedia.desc",
+        "Wikipedia"
+    );
+
+    ProtobufInputFormat inputFormat = new ProtobufInputFormat(null, protobufBytesDecoder);
+
+    KafkaSupervisorSpec supervisorSpec = createKafkaSupervisor(dataSource, dataSource, inputFormat);
+
+    final Map<String, String> startSupervisorResult = cluster.callApi().onLeaderOverlord(
+        o -> o.postSupervisor(supervisorSpec)
+    );
+    Assertions.assertEquals(Map.of("id", dataSource), startSupervisorResult);
+
+    waitForDataAndVerifyIngestedEvents(dataSource, recordCount);
+    stopSupervisor(dataSource);
+  }
+
+  @Test
+  @Timeout(30)
+  public void test_indexKafka_protobufDataFormatWithSchemaRegistry_withParser()
+  {
+    kafkaServer.createTopicWithPartitions(dataSource, 3);
+    EventSerializer serializer = new ProtobufSchemaRegistryEventSerializer(StringUtils.format("%s:%s", cluster.getEmbeddedHostname().toString(), schemaRegistry.getContainer().getMappedPort(9081)));
+    serializer.initialize(dataSource);
+    int recordCount = generateStreamAndPublishToKafka(dataSource, serializer, true);
+    SchemaRegistryBasedProtobufBytesDecoder protobufBytesDecoder = new SchemaRegistryBasedProtobufBytesDecoder(
+        StringUtils.format("http://%s:%s", cluster.getEmbeddedHostname().toString(), schemaRegistry.getContainer().getMappedPort(9081)),
+        null,
+        null,
+        null,
+        null,
+        overlord.bindings().jsonMapper()
+    );
+
+    JSONParseSpec parseSpec = new JSONParseSpec(
+        new TimestampSpec("timestamp", "auto", null),
+        createWikipediaDimensionsSpec(),
+        new JSONPathSpec(true, null),
+        null,
+        false
+    );
+
+    ProtobufInputRowParser parser = new ProtobufInputRowParser(parseSpec, protobufBytesDecoder, null, null);
+    KafkaSupervisorSpec supervisorSpec = createKafkaSupervisorWithParser(dataSource, dataSource, parser);
+
+    final Map<String, String> startSupervisorResult = cluster.callApi().onLeaderOverlord(
+        o -> o.postSupervisor(supervisorSpec)
+    );
+    Assertions.assertEquals(Map.of("id", dataSource), startSupervisorResult);
+
+    waitForDataAndVerifyIngestedEvents(dataSource, recordCount);
+    stopSupervisor(dataSource);
+  }
+
+  @Test
+  @Timeout(30)
+  public void test_indexKafka_protobufDataFormatWithSchemaRegistry()
+  {
+    kafkaServer.createTopicWithPartitions(dataSource, 3);
+    EventSerializer serializer = new ProtobufSchemaRegistryEventSerializer(StringUtils.format("%s:%s", cluster.getEmbeddedHostname().toString(), schemaRegistry.getContainer().getMappedPort(9081)));
```

Comment:
This method can also move to `KafkaSchemaRegistryResource` as `getRegistryHostAndPort()`